docs/core/deploying/preparing-libraries-for-trimming.md

```diff
@@ -0,0 +1,163 @@
+
+## Resolving trim warnings
+
+The above steps will produce warnings about code that may cause problems when used in a trimmed app. Here are a few examples of the most common kinds of warnings you may encounter, with recommendations for fixing them.
+
+### RequiresUnreferencedCode
+
+```csharp
+using System.Diagnostics.CodeAnalysis;
+
+public class MyLibrary
+{
+    public static void Foo()
+    {
+        // warning IL2026 : MyLibrary.Foo: Using method 'MyLibrary.Bar' which has
+        // 'RequiresUnreferencedCodeAttribute' can break functionality
+        // when trimming application code.
+        Bar();
+    }
+
+    [RequiresUnreferencedCode("Bar does something incompatible with trimming.")]
+    static void Bar()
+    {
+    }
+}
+```
+
+This means the library calls a method which has explicitly been annotated as incompatible with trimming, using [`RequiresUnreferencedCodeAttribute`](
+https://docs.microsoft.com/dotnet/api/system.diagnostics.codeanalysis.requiresunreferencedcodeattribute?view=net-5.0). To get rid of the warning, consider whether `Foo` needs to call `Bar` to do its job. If so, annotate the caller `Foo` with `RequiresUnreferencedCode` as well; this will "bubble up" the warning so that callers of `Foo` get a warning instead. Once you have "bubbled up" the attribute all the way to public APIs (so that these warnings are produced only for public methods, if at all), you are done. Apps which call your library will now get warnings if they call those public APIs, but these will no longer produce warnings like `IL2104: Assembly 'MyLibrary' produced trim warnings`.
+
+### DynamicallyAccessedMembers
+
+```csharp
+using System.Diagnostics.CodeAnalysis;
+
+public class MyLibrary
+{
+    public static void Foo(Type type)
+    {
+        // warning IL2067: MyLibrary.Foo(Type): 'type' argument does not satisfy
+        // 'DynamicallyAccessedMemberTypes.PublicMethods' in call to 'MyLibrary.UseMethods(Type)'.
+        // The parameter 't' of method 'MyLibrary.Foo(Type)' does not have matching annotations.
+        UseMethods(type);
+    }
+
+    static void UseMethods(
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
+        Type type)
+    {
+        foreach (var method in type.GetMethods())
+        {
+            // ...
+        }
+    }
+}
+```
+
+Here, `Foo` is calling a method which takes a `Type` argument that is annotated with a [`DynamicallyAccessedMembers`](https://docs.microsoft.com/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-5.0) requirement. The requirement states that the value passed in as this argument must represent a type whose public methods are available. In this case, you can fix this by adding the same requirement to the parameter of `Foo`. Like above, once you have bubbled up such warnings to public APIs, you are done.
```

Comment:
Maybe we should put another piece of code with the code changes for this and the other examples a before/after to be more explicit.
Also, I'm not comfortable with using "Like above, once you have bubbled ..." If we update the document we might point to something else, better to put something like "Like with RequiresUnreferencedCode, once you have bubbled ..."